### DISCUSSION TITLE
when i defined directive use defineNuxtPlugin, i got a mistake with defineNuxtPlugin is undefined

### DISCUSSION THREAD

**Discussion by JiatLn** [nuxt/nuxt#16332] [Questions] [answered]
when i defined directive use `defineNuxtPlugin`, i got a mistake with `defineNuxtPlugin` is undefined.

- src\plugins\directive.ts
```ts
export default defineNuxtPlugin(nuxtApp =>
  nuxtApp.vueApp.directive('focus', {
    mounted(el: HTMLInputElement) {
      setTimeout(() => {
        el.focus()
      }, 1000)
    },
  }),
)
```

here is the doc link: https://nuxt.com/docs/guide/directory-structure/plugins#vue-directives

**Comment by wobsoriano** [comment, marked as answer]
Did you start your server or npm install

**Reply by JiatLn** [reply]
yes, you are right.
